digdag-tests/src/test/java/acceptance/WaitIT.java

```diff
@@ -0,0 +1,120 @@
+package acceptance;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import utils.CommandStatus;
+
+import java.nio.file.Path;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.function.Supplier;
+
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.lessThan;
+import static org.junit.Assert.assertThat;
+import static utils.TestUtils.copyResource;
+import static utils.TestUtils.main;
+
+public class WaitIT
+{
+    @Rule
+    public TemporaryFolder folder = new TemporaryFolder();
+
+    private Path root()
+    {
+        return folder.getRoot().toPath().toAbsolutePath();
+    }
+
+    private static class ExecResult
+    {
+        CommandStatus commandStatus;
+        Duration duration;
+
+        public ExecResult(CommandStatus commandStatus, Duration duration)
+        {
+            this.commandStatus = commandStatus;
+            this.duration = duration;
+        }
+    }
+
+    private ExecResult runAndMonitorDuration(Supplier<CommandStatus> task)
+    {
+        Instant start = Instant.now();
+        CommandStatus commandStatus = task.get();
+        Duration duration = Duration.between(start, Instant.now());
+        return new ExecResult(commandStatus, duration);
+    }
+
+    private void testWorkflow(String workflowName, int expectedDuration)
+            throws Exception
+    {
+        String nowaitResourcePath = "acceptance/wait/nowait.dig";
+        String targetResourcePath = "acceptance/wait/" + workflowName;
+
+        Duration baselineDuration;
+        {
+            copyResource(nowaitResourcePath, root().resolve("wait.dig"));
+            ExecResult result = runAndMonitorDuration(() ->
+                    main("run", "-o", root().toString(), "--project", root().toString(), "wait.dig"));
+            CommandStatus status = result.commandStatus;
+            assertThat(status.errUtf8(), status.code(), is(0));
+            baselineDuration = result.duration;
+        }
+
+        {
+            copyResource(targetResourcePath, root().resolve("wait.dig"));
+            ExecResult result = runAndMonitorDuration(() ->
+                    main("run", "-o", root().toString(), "--project", root().toString(), "wait.dig"));
+            CommandStatus status = result.commandStatus;
+            assertThat(status.errUtf8(), status.code(), is(0));
+            assertThat(result.duration, greaterThan(baselineDuration));
+            assertThat(result.duration, lessThan(
+                    // Actual wait duration can be longer than the specified 10 seconds for some reason
+                    baselineDuration.plusSeconds((long) (expectedDuration * 1.5))));
```

Comment:
I have a bit concern in this condition.
When CI performance is degraded, it may fail unexpectedly.

Reply:
Agreed. Just tweaked the value since I don't think there is an easy robust way to handle the concern.